### PR TITLE
Erc20

### PR DESCRIPTION
**Pull request by willghatch**

### CODE REVIEW

File: examples/ERC20/index.rsh [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+const ERC165 = mixin({
+  IDs: [ Bytes.fromHex('0x01ffc9a7'), ],
+  View: [{
+    supportsInterface: Fun([Bytes(4)], Bool),
+  }],
+});
+
+const ERC20 = mixin({
+  //IDs: [ Bytes.fromHex('0xTODO'), ],
```
Comment (jeapostrophe) [inline examples/ERC20/index.rsh:16]: !

File: examples/ERC20/index.rsh [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  View: [{
+    name: Fun([], StringDyn),
+    symbol: Fun([], StringDyn),
+    // TODO Decimals should return uint8
```
Comment (jeapostrophe) [inline examples/ERC20/index.rsh:20]: 💩

File: examples/ERC20/index.rsh [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  const balances = new Map(Address, UInt);
+  const allowances = new Map(Tuple(Address, Address), UInt);
+
+  balances[D] = totalSupply;
+  E.Transfer(zeroAddress, D, totalSupply);
+
+  const [ ] =
+    parallelReduce([ ])
+      .define(() => {
+
```
Comment (jeapostrophe) [inline examples/ERC20/index.rsh:85]:
```suggestion
```

File: examples/ERC20/index.rsh (hunk 2 above)
Comment (willghatch) [reply examples/ERC20/index.rsh:20]: Ah, yeah, I wasn't sure about that.  After I checked the ABI encoding to check that using UInt for uint8 would be fine, I forgot to edit the comment.

File: examples/ERC20/index.rsh (hunk 1 above)
Comment (willghatch) [reply examples/ERC20/index.rsh:16]: Yeah, I'll compute that today and fix it.

File: examples/ERC20/index.mjs [hunk 4]
```diff
@@ -166,13 +161,13 @@ const test = async (ctc, expected) => {
...
-  await assertFail(approve(ctc, zeroAddr, 10));
+  await assertFail(approve(ctc, zeroAddress, 10));
 
   await approve(ctc, addr1, 100);
   // Can't transferFrom to the zero address
-  await assertFail(transferFrom(ctc1, ctc, zeroAddr, 10));
+  await assertFail(transferFrom(ctc1, ctc, zeroAddress, 10));
 
   assertEq(await ctc.name(), expected.name, "name()");
   assertEq(await ctc.symbol(), expected.symbol, "symbol()");
```
Comment (jeapostrophe) [inline examples/ERC20/index.mjs:173]: I don't see tests for the decimals

**Comment by willghatch** [comment]
FYI pushed a rebase, hopefully CI is happy now...

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**